### PR TITLE
Finalise conversion of Make to Rake

### PR DESCRIPTION
**Pull request by boffbowsh**
This suite now takes 4 seconds to run using Rake alone.

Best reviewed commit-by-commit. Work done by @benilovj.

I've added a Makefile which simply wraps rake with a couple of deprecation warnings.

### CODE REVIEW

**Comment by boffbowsh** [comment]
:+1: on the bits I didn't do.

**Comment by benilovj** [comment]
:+1: on the bits done by @boffbowsh

File: spec/tasks/validate_uniqueness_of_frontend_example_base_paths_spec.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        ]
+      }
+
+      it "succeeds without exceptions" do
+        expect { subject.invoke(examples) }.to_not raise_error
+      end
+    end
+
+    context "some examples have duplicate base_paths" do
+      before do
```
Comment (benlovell) [inline spec/tasks/validate_uniqueness_of_frontend_example_base_paths_spec.rb]: You could use the `output_to_stderr` matcher here to cut down on the `$stderr` juggling, and use compound or composable matchers around L54.

Comment (benilovj) [inline spec/tasks/validate_uniqueness_of_frontend_example_base_paths_spec.rb]: Is [that](https://github.com/alphagov/govuk-content-schemas/pull/185/files#diff-f29e01c5ef01add347b3091a65a5c899R44) better?

Comment (benlovell) [inline spec/tasks/validate_uniqueness_of_frontend_example_base_paths_spec.rb]: :heart_eyes:

**Comment by benlovell** [comment]
:+1: when :green_apple:

**Pull request merged**